etcdserver/api/v2http/client.go

```diff
@@ -761,6 +761,9 @@ func trimErrorPrefix(err error, prefix string) error {
 
 func unmarshalRequest(r *http.Request, req json.Unmarshaler, w http.ResponseWriter) bool {
 	ctype := r.Header.Get("Content-Type")
+	if strings.Contains(ctype, ";") {
```

Comment:
I have updated code ready to push, just want to first ask why this would be a better way of doing it?